### PR TITLE
Support inferred values when linting sys path

### PR DESCRIPTION
**Pull request by ericvergnaud**
## Changes
Infer value of sys.path.xxx() calls and raise advice if unresolved

### Linked issues

Progresses #1205
Replaces PR #1863

### CODE REVIEW

**Comment by ericvergnaud** [comment]
@nfx only change from #1863 is test coverage

File: tests/unit/source_code/test_dependencies.py [hunk 1]
```diff
@@ -20,55 +22,47 @@
 from tests.unit.conftest import MockPathLookup
 
 
+def dependency_resolver(notebook_resolver, path_lookup):
```
Comment (JCZuurmond) [inline tests/unit/source_code/test_dependencies.py:25]: Make this a fixture

File: src/databricks/labs/ucx/source_code/linters/imports.py [hunk 2]
```diff
@@ -211,10 +217,27 @@ def visit_call(self, node: Call):
...
+        relative = False
+        if isinstance(changed, Call):
+            if not self._match_aliases(changed.func, ["os", "path", "abspath"]):
+                return
+            relative = True
+            changed = changed.args[0]
+        try:
+            all_inferred = changed.inferred()
+            for inferred in all_inferred:
+                self._visit_inferred(changed, inferred, relative, is_append)
```
Comment (JCZuurmond) [inline src/databricks/labs/ucx/source_code/linters/imports.py:229]: This will not raise an `InferenceError`, right? I prefer to move it out of the try

**Review by JCZuurmond — CHANGES_REQUESTED** [review]
Two minor remarks

File: src/databricks/labs/ucx/source_code/linters/imports.py (hunk 2 above)
Comment (ericvergnaud) [reply src/databricks/labs/ucx/source_code/linters/imports.py:229]: That would require `all_inferred` to be available outside the try-except clause which would increase complexity

File: tests/unit/source_code/test_dependencies.py (hunk 1 above)
Comment (ericvergnaud) [reply tests/unit/source_code/test_dependencies.py]: done

**Pull request merged**